### ISSUE TITLE
`ElseCaseInsteadOfExhaustiveWhen` false negative when the subject is declared in the `when` parentheses

### ISSUE THREAD

**Issue by Rattenkrieg** [detekt/detekt#9607]
## Expected Behavior

The rule should report every `when` that has an `else` branch over an enum, sealed or boolean subject. The form of the subject should not matter. Kotlin allows two forms:

```kotlin
when (color) { ... }
when (val c = color) { ... }
```

The subject has the same type in both forms. Both should give this finding:

`When expression with enum class subject should not contain an else case.`

## Observed Behavior

The rule reports the first form only. The subject-variable form gives zero findings, and the `detektMain` task passes.

The same runs also flag `Variable 'c' is unused` at line 13. Analysis therefore read the subject-variable `when`. This rule alone skips it.

## Steps to Reproduce

Enable the rule. Run it over the file below. The rule needs type resolution, so use `detektMain` and not `detekt`.

Config:

```yaml
potential-bugs:
  ElseCaseInsteadOfExhaustiveWhen:
    active: true
    ignoredSubjectTypes: []
```

Reproducer:

```kotlin
enum class Color {
    RED,
    GREEN,
}

fun reported(color: Color): Int =
    when (color) {
        Color.RED -> 1
        else -> error("unhandled")
    }

fun notReported(color: Color): Int =
    when (val c = color) {
        Color.RED -> 1
        else -> error("unhandled")
    }
```

Expected: two findings, one per function.
Observed: one finding, on `reported` only.

## Your Environment

* Version of detekt used: I reproduced the bug on `2.0.0-alpha.3`, on `2.0.0-alpha.6`, on `main-SNAPSHOT` (`main-20260813.103658-670`) and on the latest stable `1.23.8`. Every run used the Gradle plugin and the `detektMain` task with analysis enabled. Every run gave one `ElseCaseInsteadOfExhaustiveWhen` finding, on `reported` at line 7. No run gave a finding on `notReported` at line 13.
* Version of Gradle used: 9.5.1 with Kotlin 2.4.10 for the 2.x runs. The `1.23.8` run needed Gradle 8.12.1 with Kotlin 2.0.21. JDK 21 throughout.
* Operating System and version: macOS 26.5.2.
* Link to your project: not a public repository. The reproducer above is self-contained.
